### PR TITLE
fix(equalizer): add input validation and named constants

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Addresses review feedback from #1991:

- **Input validation**: `NewBandPass`, `NewBandReject`, and `NewPeaking` now validate `frequency > 0` and `widthHz > 0`, returning errors instead of producing NaN filter coefficients from division by zero
- **Named constants**: Replace magic numbers `1.0` and `0.01` in `hzToOctaves` with descriptive constants (`minFrequencyHz`, `minBandEdgeHz`, `minHalfWidthHz`) per project coding guidelines
- **Tests**: Add validation error path tests for all three constructors

## Test plan

- [x] `go test -race -v ./internal/myaudio/equalizer/` — all pass
- [x] `golangci-lint run -v ./internal/myaudio/equalizer/` — zero issues
- [ ] CI passes

## Summary by CodeRabbit

* **Bug Fixes**
  * Strengthened input validation for audio filter operations with improved parameter boundary checks and error handling.
  * Updated frequency and bandwidth calculations with refined minimum value constraints.

* **Tests**
  * Expanded test coverage with table-driven subtests targeting edge cases and invalid parameter scenarios.

### CODE REVIEW

File: internal/myaudio/equalizer/equalizer_test.go [hunk 1]
```diff
@@ -154,17 +154,61 @@ func TestNewHighPass(t *testing.T) {
...
+		assert.Nil(t, f)
+		assert.ErrorContains(t, err, "frequency")
+	})
+
+	t.Run("zero_widthHz", func(t *testing.T) {
+		f, err := NewPeaking(48000, 1000, 0, 6.0, 1)
+		require.Error(t, err)
+		assert.Nil(t, f)
+		assert.ErrorContains(t, err, "widthHz")
+	})
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/myaudio/equalizer/equalizer_test.go:217]: Fixed. Added negative_frequency and negative_widthHz test cases to both TestNewPeaking and TestNewBandReject.

File: internal/myaudio/equalizer/equalizer_test.go [hunk 2]
```diff
@@ -242,6 +286,20 @@ func TestNewBandReject(t *testing.T) {
...
+		assert.Nil(t, f)
+		assert.ErrorContains(t, err, "frequency")
+	})
+
+	t.Run("zero_widthHz", func(t *testing.T) {
+		f, err := NewBandReject(48000, 1000, 0, 1)
+		require.Error(t, err)
+		assert.Nil(t, f)
+		assert.ErrorContains(t, err, "widthHz")
+	})
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/myaudio/equalizer/equalizer_test.go:320]: Fixed. Added negative_frequency and negative_widthHz test cases to TestNewBandReject.
